tests/fs/test_mmap.c

```diff
@@ -176,5 +176,53 @@ int main() {
     printf("/yolo/sharedoffset.txt content=%s %d\n", buffer + offset, offset);
     fclose(fd);
   }
+
+  /**
+   * MMAP to and expanded file
```

Comment:
Is "expanded file" terminology that is used elsewhere?  How about "over-allocated memfs file"?

Reply:
I chose the term `expanded file` because the "expansion" is done by a method called `expandFileStorage` which you can see here:

https://github.com/emscripten-core/emscripten/blob/0e4bb329e5eba4e7b16e3ceaaa0aae7a09e257ad/src/library_memfs.js#L115

Reply:
To me `expanded file` doesn't hold the right meaning.    The `expandFileStorage` may or may not over-allocate, right?  But its the fact the you over-allocate the storage that is key here IIUC, no?      I could be wrong.. I didn't write the memfs code so maybe `expanded file` means more to the authors of this code.

Reply:
when appending to a file `expandFileStorage` will increase the buffer size for that file more than the bytes requested to be written. This is to avoid having to increase the buffer for every call to write (which would be very slow). So instead the actual file size is tracked in a separate property called `usedBytes`.

mmap allocates memory based on the `usedBytes`, and calls javascripts ArrayBuffer `set` method on the heap with the files buffer as parameter. The `set` method has no parameter for the length, and so it copies the full contents of the file buffer. To avoid copying more than the actual file size the file buffer is sliced before calling `set`, but before this fix, slicing would only occur if the requested length was less than used bytes (which is mostly not the case if mapping the entire file). So instead the criteria for slicing must be if the requested length is less than the length of the file buffer - which is normally more than `usedBytes` for `expanded files`.

Reply:
but yes - `expandFileStorage` does over-allocate file storage in that sense. Just wouldn't confuse it with over-allocating of heap, since the case here is that mmap writes beyond what it allocated on the heap.

not sure what the best term here is though. Maybe we need more opinions?

Reply:
I guess you could call it "over-sized" if you prefer?  But I think over-allocated is pretty clear and applies to any object that is larger than needs to be to make room for future expansion.    We can let @kripken have the final say :)

Reply:
I don't have a strong opinion on the term, but "over-allocated" does seem pretty reasonable, let's change to that?

Also `and` should be `an` in the comment.